autopts/wid/gap.py

```diff
@@ -2226,3 +2239,91 @@ def hdl_wid_274(_: WIDParams):
     Please send L2CAP G-Frame with unicast data. Expect to perform link encryption before sending G-Frame.
     '''
     return True
+
+
+def hdl_wid_352(params: WIDParams):
+    '''
+    Please click OK when IUT establishes BIG sync, and ready to receive ISO data.
+    '''
+    addr = pts_addr_get()
+    addr_type = pts_addr_type_get()
+
+    # Do not call `btp.bap_broadcast_scan_start()` to discover peer. Because the `Broadcast_ID` is
+    # not inclued in the extended advertising report.
+    # Just calling GAP discovery to find the PTS device
+    btp.gap_start_discov(transport='le', discov_type='passive', mode='observe')
+    sleep(10)
+    btp.gap_stop_discov()
+    if not btp.check_discov_results(addr_type=addr_type, addr=addr):
+        log('Peer device not found.')
+        return False
+
+    stack = get_stack()
+
+    log('Synchronizing to broadcast')
+    btp.gap_padv_create_sync(0, 0, 0x200, 0)
+    if not stack.gap.wait_periodic_established(10):
+        log('Failed to periodic sync established')
+        return False
+
+    if params.test_case_name in ['GAP/SEC/SEM/BV-32-C']:
+        # broadcast code from TSPX_broadcast_code in ptsproject/x/gap.py
+        broadcast_code = stack.gap.big_broadcast_code
+    else:
+        broadcast_code = None
+
+    btp.gap_big_create_sync(0, 1, 1, 0, 255, broadcast_code)
+    if not stack.gap.wait_big_established(10):
+        log('BIG sync establishment failed')
+        return False
+
+    if not stack.gap.wait_bis_data_path_setup(None, 10):
+        log('Failed to setup BIS')
+        return False
+
+    return True
+
+
+def hdl_wid_351(_: WIDParams):
+    '''
+    Wait for Broadcast ISO request.
+    '''
+    return True
+
+
+def _gap_bis_broadcast(bis_id, data, data_length):
+    """
+    Broadcast BIS data. Ignore the errors when calling btp.gap_bis_broadcast().
+    """
+    try:
+        btp.gap_bis_broadcast(bis_id, data * data_length)
+    except types.BTPError:
+        log("Ignore BIS broadcast failure")
+
+
+def hdl_wid_356(_: WIDParams):
+    '''
+    Please broadcast valid ISO data packets (more than 3 packets).
+    '''
+    for _ in range(1, 100):
+        _gap_bis_broadcast(0, '00', 1)
+    return True
+
+
+def hdl_wid_355(_: WIDParams):
+    '''
+    Please verify that IUT received BIS data of 0x01, 0x02, 0x03, 0x04, 0x05, and 0x06
+    '''
+    data = get_stack().gap.read_bis_stream_received_data(None, timeout=10)
+
+    if not data:
+        log('No BIS data received')
+        return False
+
+    value = bytes([0x01, 0x02, 0x03, 0x04, 0x05, 0x06])
+    for v in data:
+        if value in v['data']:
+            return True
```

Comment:
So `data` is a list of all the data received by the BIS (or in the case of `None`, any BIS). 
Each item in this list is then another dictionary (or a named tuple?) with a `data` key? 

Wouldn't it make more sense to have `data` be a single received SDU, and when check for data in a loop? 

If by the time `data = get_stack().gap.read_bis_stream_received_data(None, timeout=10)` is called, the expected data is not received, then this will return `False`, even if the data is then received a millisecond later.

Alternative, `read_bis_stream_received_data` could be a `wait_for_data` that takes expected data as input, similar to how we wait for mostly any other event